### PR TITLE
Initiual suggestions and begining of the document

### PR DESCRIPTION
**Pull request by tmacam**
Signed-off-by: Tiago Alves Macambira <tmacam@burocrata.org>

Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [ ] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [ ] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

### CODE REVIEW

File: daprdocs/content/en/operations/extending/developing-grpc-components.md [hunk 1]
```diff
@@ -6,8 +6,45 @@ weight: 250
 description: "Extending Dapr with external gRPC-based components"
 ---
 
-[gRPC-based](https://grpc.io/) Dapr components are typically run as containers or processes that communicate with the Dapr main process via [Unix Domain Sockets](https://en.wikipedia.org/wiki/Unix_domain_socket).
+## Components and Pluggable Components
 
+Out of the box, Dapr [building blocks] come with integrations to a wide range of cloud providers and open source solutions. We call each  of these individual integrations _"components"_. For instance, a input biding integration for MySql is a _component_ and a state store integration for MySQL would be a different component.
```
Comment (mcandeia) [inline daprdocs/content/en/operations/extending/developing-grpc-components.md:11]: How about link components to the concepts page? https://docs.dapr.io/concepts/components-concept/

File: daprdocs/content/en/operations/extending/developing-grpc-components.md [hunk 2]
```diff
@@ -6,8 +6,45 @@ weight: 250
 description: "Extending Dapr with external gRPC-based components"
 ---
 
-[gRPC-based](https://grpc.io/) Dapr components are typically run as containers or processes that communicate with the Dapr main process via [Unix Domain Sockets](https://en.wikipedia.org/wiki/Unix_domain_socket).
+## Components and Pluggable Components
```
Comment (mcandeia) [inline daprdocs/content/en/operations/extending/developing-grpc-components.md:9]: looks like that both terms are interchangeably being used, can we name this section as just `Pluggable Components Overview` ?

File: daprdocs/content/en/operations/extending/developing-grpc-components.md [hunk 3]
```diff
@@ -6,8 +6,45 @@ weight: 250
...
+
+Traditionally, Dapr components are
+* written in Go,
+* are run as part of the same executable as Dapr itself,
+* integrated directly with Dapr codebase,
+* are distributed and hosted with the rest of Dapr codebase.
+
+Creating a new component for Dapr is not a huge undertaking and doing so helps not only a single developer or team but the whole Dapr community.
+
+There are circumstances for which creating a traditional Dapr component might not be feasible. For instance, for teams with no familiarity with Go. Or teams that don't want to depend on Dapr release cycle or community process for onboarding new components. Or even teams that don't want or, for a variety of reasons, can't integrate their component codebase with Dapr. For those situations there is an alternative method of extending Dapr with new functionality: gRPC pluggable components.
```
Comment (mcandeia) [inline daprdocs/content/en/operations/extending/developing-grpc-components.md:25]: capitalized version for consistency
```suggestion
There are circumstances for which creating a traditional Dapr component might not be feasible. For instance, for teams with no familiarity with Go. Or teams that don't want to depend on Dapr release cycle or community process for onboarding new components. Or even teams that don't want or, for a variety of reasons, can't integrate their component codebase with Dapr. For those situations there is an alternative method of extending Dapr with new functionality: gRPC Pluggable Components.
```

File: daprdocs/content/en/operations/extending/developing-grpc-components.md [hunk 4]
```diff
@@ -6,8 +6,45 @@ weight: 250
...
+
+### gRPC Pluggable component
```
Comment (mcandeia) [inline daprdocs/content/en/operations/extending/developing-grpc-components.md:27]:
```suggestion
### gRPC Pluggable Components
```

**Review by mcandeia — CHANGES_REQUESTED** [review]

File: daprdocs/content/en/operations/extending/developing-grpc-components.md (hunk 1 above)
Comment (tmacam) [reply daprdocs/content/en/operations/extending/developing-grpc-components.md:11]: Yes, that would be even better TBH. Having links to both "bulding blocks" and components would make things super clear and help guide those that want dive deeper into the rabbit hole of those definitions.

**Pull request merged**